### PR TITLE
promote a new testing version

### PR DESCRIPTION
**Pull request by sopos**
this PR bring two main features
* better and more consistent search for libraries
* ability to parse yaml files including main.fmf and metadata.yaml

### CODE REVIEW

**Comment by hegerj** [comment]
Hm, I see doc strings for the new yaml functions, but I can't find them in man page - they should get there automatically during build, right?
Am I building beakerlib wrong(``make && make install``)?

**Comment by hegerj** [comment]
ok, I got it to work by calling ``make all-documentation`` from ``src/`` first and then ``make && make install`` from root dir.
Is that expected workflow?

**Comment by sopos** [comment]
I would say expected workflow is `make build install` but we can change it..

File: src/rpms.sh [hunk 1]
```diff
@@ -469,7 +517,7 @@ rlCheckRequirements() {
 
 This is just a bit smarted wrapper of
 
-C<rlCheckRequirements $(rlGetMakefileRequires)>
+C<rlCheckRequirements $(rlGetYAMLdeps 'requires|recommends') || rlCheckRequirements $(rlGetMakefileRequires)>
```
Comment (hegerj) [inline src/rpms.sh:520]: "requires|recommends" should be "require|recommend"

File: src/rpms.sh [hunk 2]
```diff
@@ -497,8 +546,9 @@ rlCheckMakefileRequires() {
 
 =head3 rlAssertRequired
 
-Ensures that all Requires, specified in beakerlib-style beaker-wizard layout
-Makefile, are installed.
+Ensures that all requires and recommends specified in the metadata.yaml are installed.
+If no metadata.yaml is present a fall back to Makefile is done to check Requires
```
Comment (hegerj) [inline src/rpms.sh:550]: "fall back" shouldn't have the space in it I believe :)

File: src/rpms.sh (hunk 1 above)
Comment (sopos) [reply src/rpms.sh]: fixed by https://github.com/beakerlib/beakerlib/pull/75/commits/2731beeff30a9dc9a748a026ceb1409b08797c5f

File: src/rpms.sh (hunk 2 above)
Comment (sopos) [reply src/rpms.sh]: fixed by https://github.com/beakerlib/beakerlib/pull/75/commits/2731beeff30a9dc9a748a026ceb1409b08797c5f

**Review by hegerj — APPROVED** [review]
LGTM, thanks for the implementation.

**Pull request merged**